src/platform/src/Bridge/Failover/Tests/FailoverPlatformTest.php

```diff
@@ -0,0 +1,438 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\AI\Platform\Bridge\Failover\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+use Symfony\AI\Platform\Bridge\Failover\FailoverPlatform;
+use Symfony\AI\Platform\Exception\RuntimeException;
+use Symfony\AI\Platform\ModelCatalog\FallbackModelCatalog;
+use Symfony\AI\Platform\ModelCatalog\ModelCatalogInterface;
+use Symfony\AI\Platform\PlatformInterface;
+use Symfony\AI\Platform\Result\DeferredResult;
+use Symfony\AI\Platform\Result\InMemoryRawResult;
+use Symfony\AI\Platform\Result\TextResult;
+use Symfony\AI\Platform\Test\InMemoryPlatform;
+use Symfony\AI\Platform\Test\PlainConverter;
+use Symfony\Component\Clock\MonotonicClock;
+use Symfony\Component\RateLimiter\RateLimiterFactory;
+use Symfony\Component\RateLimiter\Storage\InMemoryStorage;
+
+final class FailoverPlatformTest extends TestCase
+{
+    public function testPlatformCannotPerformInvokeWithoutRemainingPlatform()
+    {
+        $mainPlatform = $this->createMock(PlatformInterface::class);
+        $mainPlatform->expects($this->once())->method('invoke')
+            ->willThrowException(new \Exception('The invoke method cannot be called from the main platform.'));
+
+        $failedPlatform = $this->createMock(PlatformInterface::class);
+        $failedPlatform->expects($this->once())->method('invoke')
+            ->willThrowException(new \Exception('The invoke method cannot be called from a failed platform.'));
+
+        $logger = $this->createMock(LoggerInterface::class);
+        $logger->expects($this->exactly(2))->method('error');
+
+        $failoverPlatform = new FailoverPlatform([
+            $failedPlatform,
+            $mainPlatform,
+        ], new RateLimiterFactory([
```

Comment:
lets add a static private method for creaetRateLimiterFactory() to avoid duplicate code